Panic when running the e2e tests without a compiled binary

While I could figure out why this happened, that's likely going to be hard for someone not familiar with how these tests work. Perhaps the tests could build the binary if missing. If that's a bad idea, we should just print something more informative and exit.

```
❯ go test -tags e2e ./e2e
--- FAIL: TestMergeRuleConfigWithoutLevel (0.00s)
panic: unreachable [recovered]
	panic: unreachable

goroutine 46 [running]:
testing.tRunner.func1.2({0x104b63780, 0x104c64fd0})
	/opt/homebrew/Cellar/go/1.21.0/libexec/src/testing/testing.go:1545 +0x1c8
testing.tRunner.func1()
	/opt/homebrew/Cellar/go/1.21.0/libexec/src/testing/testing.go:1548 +0x360
panic({0x104b63780?, 0x104c64fd0?})
	/opt/homebrew/Cellar/go/1.21.0/libexec/src/runtime/panic.go:914 +0x218
github.com/styrainc/regal/e2e.ExitStatus({0x104c67358?, 0x14000514120})
	/Users/anderseknert/git/styra/regal/e2e/cli_test.go:510 +0xec
github.com/styrainc/regal/e2e.TestMergeRuleConfigWithoutLevel(0x1400045ab60)
	/Users/anderseknert/git/styra/regal/e2e/cli_test.go:462 +0x15c
testing.tRunner(0x1400045ab60, 0x104c62568)
	/opt/homebrew/Cellar/go/1.21.0/libexec/src/testing/testing.go:1595 +0xe8
created by testing.(*T).Run in goroutine 1
	/opt/homebrew/Cellar/go/1.21.0/libexec/src/testing/testing.go:1648 +0x33c
FAIL	github.com/styrainc/regal/e2e	0.293s
FAIL
```

--- a/e2e/cli_test.go
+++ b/e2e/cli_test.go
@@ -5,6 +5,7 @@ package e2e
 import (
 	"bytes"
 	"encoding/json"
+	"errors"
 	"io"
 	"log"
 	"os"
@@ -465,11 +466,19 @@ func TestMergeRuleConfigWithoutLevel(t *testing.T) {
 }
 
 func binary() string {
+	location := "../regal"
+
 	if b := os.Getenv("REGAL_BIN"); b != "" {
-		return b
+		location = b
 	}
 
-	return "../regal"
+	if _, err := os.Stat(location); errors.Is(err, os.ErrNotExist) {
+		log.Fatal("regal binary not found — make sure to run go build before running the e2e tests")
+	} else if err != nil {
+		log.Fatal(err)
+	}
+
+	return location
 }
 
 func regal(outs ...io.Writer) func(...string) error {

